src/lib/ffi/ffi_oid.h

```diff
@@ -0,0 +1,20 @@
+/*
+* (C) 2017 Jack Lloyd
+* (C) 2025 Dominik Schricker
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_FFI_OID_H_
+#define BOTAN_FFI_OID_H_
+
+#include <botan/asn1_obj.h>
+#include <botan/internal/ffi_util.h>
+
+extern "C" {
+
+// TODO get a proper magic number from somewhere:tm:
+BOTAN_FFI_DECLARE_STRUCT(botan_asn1_oid_struct, Botan::OID, 0x123456);
```

Comment:
How should I create / compute the magic number here?

Reply:
Any random 32-bit integer (eg `botan rng --format=hex 4`). Main point is that it does not collide with any other FFI magic number, so someone accidentally using the wrong type gets a reasonable error rather than a memory corruption. Secondary point is tht it isn't likely otherwise to collide with memory in a random process stack (eg all 0s is valid 32-bit integer, doesn't collide with other uses, but can easily collide with something randomly on a heap)